### ISSUE TITLE
Problem: Upgrade from 1.11.2 to 1.12 on RHEL7 fails to create Django static files

### ISSUE THREAD

**Issue by uofmsean** [archivematica/Issues#1312]
**Expected behaviour**
Should be no Python errors during "yum update" step during upgrade. Should not have to manually run "manage.py collectstatic" afterwards.

**Current behaviour**
Received Python error related to RemovedInDjango19Warning during the "manage.py collectstatic --noinput --clear" part of the upgrade for both dashboard and storage-service. This may be related to the Django version update to 1.11.29 but the older 1.8.19 being used to run manage.py for some reason. The failure to run collectstatic causes the Dashboard service to fail to start (although the Storage Service did start).
```
Full trace for storage service during upgrade follows (dashboard is similar):
  File "manage.py", line 9, in <module>
    from django.core.management import execute_from_command_line
  File "/usr/share/archivematica/virtualenvs/archivematica-storage-service/lib/python2.7/site-packages/django/core/management/__init__.py", line 10, in <module>
    from django.apps import apps
  File "/usr/share/archivematica/virtualenvs/archivematica-storage-service/lib/python2.7/site-packages/django/apps/__init__.py", line 1, in <module>
    from .config import AppConfig
  File "/usr/share/archivematica/virtualenvs/archivematica-storage-service/lib/python2.7/site-packages/django/apps/config.py", line 6, in <module>
    from django.utils.module_loading import module_has_submodule
  File "/usr/share/archivematica/virtualenvs/archivematica-storage-service/lib/python2.7/site-packages/django/utils/module_loading.py", line 4, in <module>
    from importlib import import_module
  File "/usr/share/archivematica/virtualenvs/archivematica-storage-service/lib/python2.7/site-packages/django/utils/importlib.py", line 6, in <module>
    from django.utils.deprecation import RemovedInDjango19Warning
ImportError: cannot import name RemovedInDjango19Warning
```

The issue can be corrected by running the following (as root) after the "yum update" and before starting the Dashboard and Storage Service:

```
bash -c " \
    set -a -e -x
    source /etc/default/archivematica-dashboard || \
        source /etc/sysconfig/archivematica-dashboard \
            || (echo 'Environment file not found'; exit 1)
    cd /usr/share/archivematica/dashboard
    /usr/share/archivematica/virtualenvs/archivematica-dashboard/bin/python manage.py collectstatic --noinput --clear
";
bash -c " \
    set -a -e -x
    source /etc/default/archivematica-storage-service || \
        source /etc/sysconfig/archivematica-storage-service \
            || (echo 'Environment file not found'; exit 1)
    cd /usr/lib/archivematica/storage-service
    /usr/share/archivematica/virtualenvs/archivematica-storage-service/bin/python manage.py collectstatic --noinput --clear
";
```

**Steps to reproduce**
Upgrade from 1.11.2 to 1.12 on RHEL7 (or CentOS7).

**Your environment (version of Archivematica, operating system, other relevant details)**
Upgrade starting from Archivematica 1.11.2 and Storage Service 0.16.1 and Django 1.8.19
Python 2.7.5
Red Hat Enterprise Linux Server release 7.8 (Maipo)
Linux 3.10.0-1127.18.2.el7.x86_64 #1 SMP Mon Jul 20 22:32:16 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

---

**For Artefactual use:**

Before you close this issue, you must check off the following:

- [ ] All pull requests related to this issue are properly linked
- [ ] All pull requests related to this issue have been merged
- [ ] A testing plan for this issue has been implemented and passed (testing plan information should be included in the issue body or comments)
- [ ] Documentation regarding this issue has been written and merged (if applicable)
- [ ] Details about this issue have been added to the release notes (if applicable)

**Comment by mamedin** [comment]
Thanks @uofmsean for details and workaround :) . This workaround is already included in the packages as post install scripts:

https://github.com/artefactual-labs/am-packbuild/blob/qa/1.x/rpm/archivematica/archivematica.spec#L277-L281
https://github.com/artefactual-labs/am-packbuild/blob/qa/1.x/rpm/archivematica-storage-service/package.spec#L102-L106

I'm going to try to reproduce the issue.

**Comment by uofmsean** [comment]
The first indicator that there was a problem was the error from the "yum upgrade" (for both the dashboard and storage-service), and then when the dashboard wouldn't start, that led me to the static file issue. I did review the post install scripts from the RPMs for the dashboard and storage-service, and that's what prompted me to re-run the collectstatic part. The commands worked fine when run manually post-upgrade, but failed initially during the post install phase of the upgrade. Based on what @sevein mentioned, something from the older Django install seemed to get in the way but only during the post install phase.

**Comment by mamedin** [comment]
The faster workaround while the ongoing PR is merged and new packages published is:

```
sudo yum reinstall archivematica*
```

The reason of this issue is because the `manage.py collectstatic` command is run before the `%postun` build script. This is the output of the `%postun` build script:

```
Cleaning up / removing...
   2:archivematica-dashboard-1.11.2-1 ################################# [100%]
```

The `%postun` build script removes the obsolete files from old package (virtualenv files included), so it needs to be run just before the `manage.py collectstatic` command. In this case the virtualenv was using the old django version when upgrading the package (`ImportError: cannot import name RemovedInDjango19Warning`).

So, makes sense the package is installed without errors when reinstalling because the `%postun` task was already applied. Same when running the  `manage.py collectstatic` command from CLI. tt worked fine on ansible deploys with rpm packages because the `manage.py collectstatic` command is run in a separate task after upgrading packages.

I tested building new packages with the `manage.py collectstatic` command in the `%posttrans` build script intead of `%post` build script and it fisex the issue. (See https://docs.fedoraproject.org/en-US/packaging-guidelines/Scriptlets/#ordering )

**Comment by dongourley** [comment]
I would like to report that I had very similar problem updating (by package) from 1.11.2 to 1.12 on Ubuntu 18. In my case the `apt-get upgrade` seemed to complete successfully, but when I try to access the dashboard I get a 500 internal server error and the dashboard.log has a bunch of these:
`ValueError: Missing staticfiles manifest entry for 'images/favicon.ico'`

I tried running "manage.py collectstatic" manually and that copied a bunch of stuff until it got to font-awesome when it reported:
`IOError: [Errno 2] No such file or directory: u'/usr/share/archivematica/dashboard/media/vendor/font-awesome'` 

Let me know if this is something different that I should report separately.

**Comment by jens-st** [comment]
I can confirm that. I tried to test 1.12 and run into the same issue on 18.04 (bionic beaver), both on upgrading from 1.11.2 or on a fresh install of 1.12 (using the documentation for manual upgrade or installation).
You get a "Internal Server Error" page instead of the dashboard and the dashboard logs report a repeating Django error:
```
ERROR     2020-10-30 13:50:02  django.request:exception:handle_uncaught_exception:135:  Internal Server Error: /administration/accounts/login/
Traceback (most recent call last):
  File "/usr/share/archivematica/virtualenvs/archivematica-dashboard/local/lib/python2.7/site-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/usr/share/archivematica/virtualenvs/archivematica-dashboard/local/lib/python2.7/site-packages/whitenoise/middleware.py", line 25, in __call__
    response = self.get_response(request)
  File "/usr/share/archivematica/virtualenvs/archivematica-dashboard/local/lib/python2.7/site-packages/django/core/handlers/exception.py", line 43, in inner
    response = response_for_exception(request, exc)
  File "/usr/share/archivematica/virtualenvs/archivematica-dashboard/local/lib/python2.7/site-packages/django/core/handlers/exception.py", line 93, in response_for_exception
    response = handle_uncaught_exception(request, get_resolver(get_urlconf()), sys.exc_info())
  File "/usr/share/archivematica/virtualenvs/archivematica-dashboard/local/lib/python2.7/site-packages/django/core/handlers/exception.py", line 143, in handle_uncaught_exception
    return callback(request, **param_dict)
  File "/usr/share/archivematica/virtualenvs/archivematica-dashboard/local/lib/python2.7/site-packages/django/utils/decorators.py", line 149, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/usr/share/archivematica/virtualenvs/archivematica-dashboard/local/lib/python2.7/site-packages/django/views/defaults.py", line 76, in server_error
    return http.HttpResponseServerError(template.render())
  File "/usr/share/archivematica/virtualenvs/archivematica-dashboard/local/lib/python2.7/site-packages/django/template/backends/django.py", line 66, in render
    return self.template.render(context)
  File "/usr/share/archivematica/virtualenvs/archivematica-dashboard/local/lib/python2.7/site-packages/django/template/base.py", line 207, in render
    return self._render(context)
  File "/usr/share/archivematica/virtualenvs/archivematica-dashboard/local/lib/python2.7/site-packages/django/template/base.py", line 199, in _render
    return self.nodelist.render(context)
  File "/usr/share/archivematica/virtualenvs/archivematica-dashboard/local/lib/python2.7/site-packages/django/template/base.py", line 990, in render
    bit = node.render_annotated(context)
  File "/usr/share/archivematica/virtualenvs/archivematica-dashboard/local/lib/python2.7/site-packages/django/template/base.py", line 957, in render_annotated
    return self.render(context)
  File "/usr/share/archivematica/virtualenvs/archivematica-dashboard/local/lib/python2.7/site-packages/django/template/loader_tags.py", line 177, in render
    return compiled_parent._render(context)
  File "/usr/share/archivematica/virtualenvs/archivematica-dashboard/local/lib/python2.7/site-packages/django/template/base.py", line 199, in _render
    return self.nodelist.render(context)
  File "/usr/share/archivematica/virtualenvs/archivematica-dashboard/local/lib/python2.7/site-packages/django/template/base.py", line 990, in render
    bit = node.render_annotated(context)
  File "/usr/share/archivematica/virtualenvs/archivematica-dashboard/local/lib/python2.7/site-packages/django/template/base.py", line 957, in render_annotated
    return self.render(context)
  File "/usr/share/archivematica/virtualenvs/archivematica-dashboard/local/lib/python2.7/site-packages/django/templatetags/static.py", line 105, in render
    url = self.url(context)
  File "/usr/share/archivematica/virtualenvs/archivematica-dashboard/local/lib/python2.7/site-packages/django/templatetags/static.py", line 102, in url
    return self.handle_simple(path)
  File "/usr/share/archivematica/virtualenvs/archivematica-dashboard/local/lib/python2.7/site-packages/django/templatetags/static.py", line 117, in handle_simple
    return staticfiles_storage.url(path)
  File "/usr/share/archivematica/virtualenvs/archivematica-dashboard/local/lib/python2.7/site-packages/django/contrib/staticfiles/storage.py", line 162, in url
    return self._url(self.stored_name, name, force)
  File "/usr/share/archivematica/virtualenvs/archivematica-dashboard/local/lib/python2.7/site-packages/django/contrib/staticfiles/storage.py", line 141, in _url
    hashed_name = hashed_name_func(*args)
  File "/usr/share/archivematica/virtualenvs/archivematica-dashboard/local/lib/python2.7/site-packages/django/contrib/staticfiles/storage.py", line 432, in stored_name
    raise ValueError("Missing staticfiles manifest entry for '%s'" % clean_name)
ValueError: Missing staticfiles manifest entry for 'images/favicon.ico'
```
